### DISCUSSION TITLE
限流问题

### DISCUSSION THREAD

**Discussion by lf2186** [bfenetworks/bfe#852] [Q&A (问题和回答)] [answered]
hi，您好 
                       "action": {
                                "cmd": "CLOSE",
                                "params": []
                        },
                        "checkPeriod": 5,
                        "stayPeriod": 10,
                        "threshold": 1000,
                        "accessDictSize": 1000,
                        "prisonDictSize": 1000

以上是某个域名的限流配置，结果是qps超过1000，所有请求都失败了，惩罚时长是10秒，但是业务实际需求是限流1000qps，那么会一直有1000qps请求成功，超出的部分请求失败。

这个需求bfe现在的版本能实现吗？？

**Comment by mileszhang2016** [comment]
抱歉，没太看懂你的问题。能否再说一下问题和需求？

**Reply by kwanhur** [reply]
我看问题意思是上述配置后，单域名服务访问超过1000QPS后，在惩罚周期内，所有请求均被丢弃

用户真实需求是在惩罚周期内，依然有1000QPS请求可通过，超过部分的请求则被拦截

@lf2186 请附上最小复现配置样例 :-)

**Comment by iyangsj** [comment, marked as answer]
@lf2186 可以将惩罚周期stayPeriod设置为0：

"checkPeriod": 1,  // 检测周期为1秒
"stayPeriod": 0, // 额外惩罚时长为0
"threshold": 1000, // 限流阈值为1000请求

**Reply by lf2186** [reply]
对的，现在stayPeriod改为0了。大概测下来是，checkPeriod：5，threshold：1000，那就是5秒内允许1000个请求；checkPeriod：1，threshold：1000，那是1秒内允许有1000个请求。
